example/linear2d.f90

```diff
@@ -0,0 +1,29 @@
+program linear2d_example
+
+  use nf, only: input, network, sgd, linear2d, mse, flatten
+  implicit none
+
+  type(network) :: net
+  real :: x(3, 4) = reshape( &
+    [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2], &
+    [3, 4])
+  real :: y(3) = [0.12, 0.1, 0.3]
+  integer, parameter :: num_iterations = 500
+  integer :: n
+  
+  net = network([ &
+    input(3, 4), &
+    linear2d(3, 4, 1), &
+    flatten() &
+  ])
+  
+  call net % print_info()
+
+  do n = 1, num_iterations
+    call net % forward(x)
+    call net % backward(y, mse())
+    call net % update(optimizer=sgd(learning_rate=1.))
+    print '(i4,3(3x,f8.6))', n, net % predict(x)
+  end do
```

Comment:
Should this program not be converging toward `y` values? Why doesn't it?

Reply:
Because I took random values with absurdly high learning rate. As you can see, it immediately jumps over any extremums. I mostly used the example to make sure that it conforms to SOTA implementations.
I will get it to converge